lib/discourse_chat_integration/provider/matrix/matrix_provider.rb

```diff
@@ -88,6 +88,18 @@ def self.trigger_notification(post, channel, rule)
           end
         end
       end
+
+      def self.get_channel_by_name(name)
```

Comment:
For all of these, I wonder if you should have this be a mixin somehow where you set some attrebute on the class like `channel_name_key` = "name" (or "identifier"), and then the mixin defines 
```
def self.get_channel_by_name(name)
        DiscourseChatIntegration::Channel
          .with_provider(PROVIDER_NAME)
          .with_data_value(channel_name_key_here, name)
          .first
      end

      # used in the MigrateTagAddedFilterToAllProviders migration
      def self.get_channel_name(channel)
        channel.data[channel_name_key_here]
      end
 ```

Reply:
Resolved -- since everything is a module it gets tricky to do a clean mixin and it's better to keep these defined separately.